### ISSUE TITLE
Anchor in swagger yaml file not being processed

### ISSUE THREAD

**Issue by ilinamitra** [go-swagger/go-swagger#1193]
## Problem statement

Hi there,

I'm trying to use an anchor in my swagger file so I can have multiple parameters be shared across all paths. I am aware that parameters can be defined in the global parameters section and be referenced multiple places, but I'd rather use an anchor so I can just reference the list of parameters that I want to place in all my paths (rather than repeating the reference to each parameter for each path).

The following works:
```
parameters:
  RequestIdParam:
    name: X-Request-ID
    in: header
    type: string

paths:
  /objects:
      parameters:
      - $ref: '#/parameters/RequestIdParam'
      get:
        summary: "Lists the objects"
        ...
```

However this does not work:
```
parameters:
  RequestIdParam:
    name: X-Request-ID
    in: header
    type: string

DefaultParameters: &DEFAULT_HEADERS
  parameters:
  - $ref: '#/parameters/RequestIdParam'

/objects:
    <<: *DEFAULT_HEADERS
    get:
      summary: "Lists the objects"
      ...
```

I'm confused as to why the merge is not picking up the path-level parameters. Am I missing something obvious here?

**Comment by casualjim** [comment]
This should work because the yaml library does this during parsing. So you are still sane :)
However swagger ignores those extra fields so perhaps do 

parameters:
  RequestIdParam:
    name: X-Request-ID
    in: header
    type: string

x-DefaultParameters: &DEFAULT_HEADERS
  parameters:
  - $ref: '#/parameters/RequestIdParam'

paths:
  /objects:
      <<: *DEFAULT_HEADERS
      get:
        summary: "Lists the objects"

**Comment by ilinamitra** [comment]
I gave that a try, but alas it does not work :( Any other suggestions?

**Comment by ilinamitra** [comment]
Interestingly enough, I was able to get the following to work:

```
parameters:
  RequestIdParam:
    name: X-Request-ID
    in: header
    type: string

x-DefaultParameters: 
  parameters: &DEFAULT_HEADERS
  - $ref: '#/parameters/RequestIdParam'

paths:
  /objects:
      parameters: *DEFAULT_HEADERS
      get:
        summary: "Lists the objects"
```

So using the alias seems to work fine, it's just using merge that has issues.

**Comment by EleanorRigby** [comment]
@ilinamitra  : Seems like this issue is there in yaml v2 go package (https://github.com/go-yaml/yaml/issues/184)

**Comment by fredbi** [comment]
Tested with latest version of go-yaml/yaml package and still the same. Monitoring go-yaml/yaml#184 so we may anchor slices in specs.

**Comment by cup** [comment]
https://github.com/drone/drone/issues/2323
https://github.com/drone/drone-cli/pull/103

https://github.com/go-swagger/go-swagger/issues/1193
https://github.com/go-yaml/yaml/issues/184
https://github.com/go-yaml/yaml/issues/344
https://github.com/go-yaml/yaml/issues/91
https://github.com/mikefarah/yq/issues/178
https://github.com/mikefarah/yq/issues/206

**Comment by powerman** [comment]
Looks like it won't be fixed in gopkg.in/yaml.v2, so maybe it's worth to switch to gopkg.in/yaml.v3 or https://github.com/buildkite/yaml (v2 fork patched for this issue)?

**Comment by fredbi** [comment]
@powerman  yes I've seen that yaml.v3 is out, but couldn't test it yet. Besides this particular anchoring issue, yaml.v3 brings a lot of changes.

**Comment by fredbi** [comment]
@powerman I believe the only practical way would be to migrate to go-yaml v3.
I think it preferable to wait for proper stable tagging of this new branch: go-yaml/yaml#487.

**Comment by dschro-1993** [comment]
Any updates?
